utility/include/isobus/utility/thread_synchronization.hpp

```diff
@@ -88,16 +98,368 @@ class LockFreeQueue
 	std::queue<T> queue; ///< The queue
 };
 
+#elif defined USE_CMSIS_RTOS2_THREADING
+
+#include "cmsis_os.h"
+
+#include <atomic>
+#include <cstdint>
+#include <functional>
+#include <vector>
+
+namespace isobus
+{
+	/// @brief A wrapper around a CMSIS RTOS 2 mutex.
+	/// @details See definition at https://www.keil.com/pack/doc/CMSIS/RTOS2/html/group__CMSIS__RTOS.html
+	class Mutex
```

Comment:
thought: Mutex is not a template class, so we should separate the implementation and definition?